samples/match/match_enum_else_binding.jakt

```diff
@@ -0,0 +1,46 @@
+/// Expect:
+/// - output: "I found an A 1\n1\n2\n3\n3\n3\n"
+
+enum Foo {
+    A(x: i64)
+    B(x: i64)
+    C(x: i64)
+}
+
+enum Bar {
+    A(y: i64)
+    B(x: i64)
+    C(x: i64)
+}
+
+function match_split(a: Bar) -> i64 => match a {
+    A(y) => y
+    else(x) => x
+}
+
+function main() {
+    let a = Foo::A(x: 1)
+
+    match a {
+        else(x) => println("Just {}", x)
+        A(x) => println("I found an A {}", x)
```

Comment:
Wouldn't this one just always print the first case? `else(x)` would bind to all the cases.

Reply:
Current behavior is that match will work irrespective of order, so you can have else first and it would still match to A, so I am keeping that same behavior for else bindings.

Reply:
I think that'd be too surprising since order matters in a match traditionally.